aiounifi/devices.py

```diff
@@ -338,3 +378,70 @@ def __getitem__(self, obj_id: int) -> Port:
     def __iter__(self) -> Iterator[int | str]:
         """Iterate over ports."""
         return iter(self.ports)
+
+
+class Outlet:
+    """Represents an outlet."""
+
+    def __init__(self, raw: dict) -> None:
+        """Initialize outlet."""
+        self.raw = raw
+
+    @property
+    def name(self) -> str:
+        """Name of outlet"""
+        return self.raw["name"]
+
+    @property
+    def index(self) -> int:
+        """Outlet index."""
+        return self.raw["index"]
+
+    @property
+    def has_relay(self) -> bool:
+        """Is the outlet controllable"""
+        return self.raw["has_relay"]
+
+    @property
+    def has_metering(self) -> bool:
+        """Is metering supported."""
+        return self.raw["has_metering"]
+
+    @property
+    def relay_state(self) -> bool:
+        """Is outlet power on."""
+        return self.raw["relay_state"]
+
+    @property
+    def cycle_enabled(self) -> bool | None:
+        """Modem Power Cycle"""
+        return self.raw.get("cycle_enabled")
```

Comment:
Last thing! It seems in all examples in the tests the cycle is either True or False, should this be more strict as well?

Either to return False if key does not exist or always expect the value to exist

Reply:
In the PLUG_UP1 test it is set to `assert outlet_1.cycle_enabled is None`.  That device does not report that property out of the box.  It is probably as safe bet that if cycle_enabled is missing it can be assumed to be False.  I'll make that change.

Reply:
Sorry missed that, yes keeping it to only bool simplifies things with regards to typing